.github/workflows/ci.yml

```diff
@@ -13,132 +13,102 @@ concurrency:
         aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
 
-    - name: Restore Nix Caches
-      id: nix-cache-restore
-      uses: actions/cache/restore@v3
+    - name: "Cache cargo"
+      uses: actions/cache@v3
       with:
         path: |
-          ~/.cache/nix
-        key: ${{ runner.os }}-nix-cache
-
-    # This is used to detect any new store paths we create while running builds
-    # so that they can be pushed to caches.
-    - name: Record Nix Store Paths
-      run: |
-        NIX_STORE_DIR="$(
-          nix eval --impure --raw --expr builtins.storeDir;
-        )";
-        echo "NIX_STORE_DIR=$NIX_STORE_DIR" >> "$GITHUB_ENV";
-        STORE_PATHS_FILE="$( mktemp; )";
-        echo "STORE_PATHS_FILE=$STORE_PATHS_FILE" >> "$GITHUB_ENV";
-        find "$NIX_STORE_DIR" -maxdepth 1 -mindepth 1 -type d -o -type l  \
-          |sort > "$STORE_PATHS_FILE";
-        echo "Have $(
-          wc -l "$STORE_PATHS_FILE"|cut -d' ' -f1;
-        ) $NIX_STORE_DIR paths" >&2;
-
-    - name: Build flox-gh
-      run: |
-        nix build '.#flox-gh' -L --no-update-lock-file      \
-                              --print-out-paths --no-link;
-
-    - name: Build nix-editor
-      run: |
-        nix build '.#nix-editor' -L --no-update-lock-file      \
-                                 --print-out-paths --no-link;
-
-    - name: Build builtfilter-rs
-      run: |
-        nix build '.#builtfilter-rs' -L --no-update-lock-file      \
-                                     --print-out-paths --no-link;
-
-    - name: Build flox-bash
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: cargo-${{ runner.os }}-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: cargo-${{ runner.os }}-
+
+    - name: "Cargo Test"
       run: |
-        nix build '.#flox-bash' -L --no-update-lock-file      \
-                                --print-out-paths --no-link;
+        nix develop -L \
+          --extra-experimental-features "nix-command flakes" \
+          --no-update-lock-file \
+          '.#flox' \
+            --command cargo test --locked --workspace;
+
+  nix-build:
+    name: "Build and Run Bats Tests"
+    runs-on: ${{ matrix.os }}
 
-    # These cannot be run offline so we need to build in our tree.
-    # This just builds the `flox' deps so that they can get cached.
-    - name: Build Development Env
-      run: nix develop '.#flox' --no-update-lock-file -c true;
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+        - "ubuntu-22.04-8core"
+        - "macos-latest"
 
-    - name: Build flox
-      run: |
-        nix build '.#flox' -L --no-update-lock-file --print-out-paths;
-        echo "FLOX_CLI=$( readlink -f ./result; )/bin/flox" >> "$GITHUB_ENV";
-        rm ./result;
+    steps:
+    - name: "Checkout"
+      uses: actions/checkout@v3
+      with:
```

Comment:
consider removing this for "faster" clones.
previously needed to facilitate `publish`